File: codes/java/Code_3_04_2.java
import java.util.*;

class Code_3_04_2 {
	public static void main(String[] args) {
		// ステップ１：入力
		Scanner sc = new Scanner(System.in);
		int N = sc.nextInt();
		int[] P = new int[N + 1];
		int[] Q = new int[N + 1];
		for (int i = 1; i <= N; i++) {
			P[i] = sc.nextInt();
			Q[i] = sc.nextInt();
		}
		
		// ステップ２：答えの計算・出力
		double answer = 0.0;
		for (int i = 1; i <= N; i++) {
			answer += (double)Q[i] / P[i];
		}
		System.out.format("%.12f\n", answer);
	}
}

File: codes/c/Code_3_04_2.c
#include <stdio.h>

int main() {
	// 入力
	int N, P[59], Q[59];
	scanf("%d", &N);
	int i;
	for (i = 1; i <= N; i++) {
		scanf("%d%d", &P[i], &Q[i]);
	}

	// 答えの計算 → 答えの出力
	double answer = 0.0;
	for (i = 1; i <= N; i++) {
		answer += 1.0 * Q[i] / P[i];
	}
	printf("%.12lf\n", answer);
	return 0;
}
